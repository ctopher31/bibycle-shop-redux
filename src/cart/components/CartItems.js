import React from 'react';

const CartItems = ({ children, clearCart }) => (
  <div className="cart--items">
    <h2>Items</h2>
    {children}
    <button onClick={() => clearCart()}>Clear Cart</button>
  </div>
);

export default CartItems;
