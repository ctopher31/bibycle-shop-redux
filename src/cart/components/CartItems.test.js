import React from 'react';
import { mount } from 'enzyme';
import CartItems from './CartItems';

describe('Cart component', () => {
  it('renders correctly', () => {
    const Children = () => <div className="children">Children</div>;
    const wrapper = mount(
      <CartItems>
        <Children />
      </CartItems>
    );
    const container = wrapper.find(CartItems);
    const children = container.find(Children);

    expect(wrapper.contains(<h2>Items</h2>)).toBe(true);
    expect(wrapper.exists('.cart--items')).toBe(true);
    expect(container.find(Children).length).toEqual(1);
    expect(children.exists('.children')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
