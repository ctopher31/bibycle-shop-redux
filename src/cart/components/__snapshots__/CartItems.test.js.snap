// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component renders correctly 1`] = `ReactWrapper {}`;
